### PR TITLE
chore: delete some unused native code

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [] 

### Description

Deletes code and assets related to unused RefinementOptionsViewController, deletes duplicate references to native modules.

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- deleted unused native code - brian

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (deleted unused native code - brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 332e311a65290119dc1e4a5d9bc23d8a0c98b704
</p>

File: ios/Podfile.lock [hunk 1]
```diff
@@ -2075,7 +2075,7 @@ SPEC CHECKSUMS:
   fmt: ff9d55029c625d3757ed641535fd4a75fedc7ce9
   Forgeries: 64ced144ea8341d89a7eec9d1d7986f0f1366250
   FXBlurView: db786c2561cb49a09ae98407f52460096ab8a44f
-  glog: 04b94705f318337d7ead9e6d17c019bd9b1f6b1b
```
Comment (brainbicycle) [inline ios/Podfile.lock:2078]: fix die glog

Comment (gkartalis) [reply ios/Podfile.lock:2078]: typical bad glog situation

**Review by gkartalis — APPROVED** [review]
<image width="400" src="https://media.giphy.com/media/l0MYI62whrIll8PCg/giphy.gif?cid=790b76111zzm2ab1h10qoaxbcaf60mn3u5j0khhnb7exhamw&ep=v1_gifs_search&rid=giphy.gif&ct=g" />

File: ios/Artsy.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -240,9 +240,6 @@
 		CBD794DB2BBC40A4003B3CB5 /* PrivacyInfo.xcprivacy in Resources */ = {isa = PBXBuildFile; fileRef = CBD794DA2BBC40A4003B3CB5 /* PrivacyInfo.xcprivacy */; };
 		CBE0D12E2C20C8D800DFF17E /* libBrazeUI.a in Frameworks */ = {isa = PBXBuildFile; fileRef = CBE0D12D2C20C8D800DFF17E /* libBrazeUI.a */; };
 		CBE2946229E4C19000AA77DD /* VirtualArtwork.swift in Sources */ = {isa = PBXBuildFile; fileRef = CBE2946129E4C19000AA77DD /* VirtualArtwork.swift */; };
-		CBE3211D27D9615000395A93 /* ARTCityGuideViewManager.m in Sources */ = {isa = PBXBuildFile; fileRef = CBE3210927D9615000395A93 /* ARTCityGuideViewManager.m */; };
```
Comment (MounirDhahri) [inline ios/Artsy.xcodeproj/project.pbxproj:243]: Was this removal required?
